clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -1101,6 +1101,56 @@ def BitfieldInfoAttr : CIR_Attr<"BitfieldInfo", "bitfield_info"> {
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// AnnotationAttr
+//===----------------------------------------------------------------------===//
+def AnnotationAttr : CIR_Attr<"Annotation", "annotation"> {
+  let summary = "Annotation attribute for global variables and functions";
+  let description = [{
+    Represent C/C++ attribute of annotate in CIR. example c code is:
+    ```
+     in test.c line 10
+     int *a __attribute__((annotate("testptr", "21", 12 ))); 
+    ```
+     this example code, the AnnotationAttr records annotation name "testptr", 
+     and arguments "21" and 12 as ArrayAttr args for global variable a. 
+  }];
+
+  // args is empty when there is no arg.
+  let parameters = (ins "StringAttr":$name,
+                        "ArrayAttr":$args);
+
+  let assemblyFormat = "`<` struct($name, $args) `>`";                      
+
+  let extraClassDeclaration = [{
+    bool isNoArgs() const { return getArgs().empty(); };
+  }];
+}
+
+//===----------------------------------------------------------------------===//
+// AnnotationValueAttr
+//===----------------------------------------------------------------------===//
+def AnnotationValueAttr : CIR_Attr<"AnnotationValue", "annotation value"> {
```

Comment:
Doesn't match what we discussed and it's not very useful as is. Things to fix:
- Should be named `GlobalAnnotationValueAttr`.
- It's only param should be a `ArrayAttr`.
- The verifier will walk the array and make sure that each element is a pair of StringAttr and AnnotateAttr.

After LoweringPrepare we'll get something like: `cir.global_annotationg = #cir.global_annotation<[["a", AnnotateAttr], ["b", AnnotateAttr], ...>`